### ISSUE TITLE
Refactor test suite

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/multidict#98]
Now dependency tree for `test_multidict.py` is too complicated ever for me.
We need to slip it into several files, rewrite tests in pytest style and reduce test class dependencies.

Like:
```python
READONLY_DICTS = [MultiDict, CIMultiDict, MultiDictProxy, CIMultiDictProxy]  # plus pure python classes

@pytest.parametrize(READONLY_DICTS)
def test_something(dct):
    pass
```

**Comment by webknjaz** [comment]
I'd add:
* [ ] eliminate use of pure `unittest` (inheriting from `TestCase` blocks using some of pytest features)
* [ ] move fixtures, which are copy-pasted across test modules to `testconfig.py`

**Comment by webknjaz** [comment]
I believe this is fully addressed with #915 and https://github.com/aio-libs/multidict/commit/7513e6a3dd531a78cbcda41c50362739472b7629.
